### DISCUSSION TITLE
Read Before Edit

### DISCUSSION THREAD

**Discussion by force2k1** [cline/cline#662] [Feature Requests]
The most frequent failure mode I still hit with Cline occurs when it makes assumptions about a file's current contents. It goes ahead and starts an 'edit' to add a new feature, but just assumes the current structure of the code without looking. 

I think it would solve a lot of problems if there was a deterministic check for cases where the file already exists, and Cline attempts an edit operation before the file has been read into the context window. If Cline tries and edit when the existing content hasn't been read in, abort and force-feed it the current file contents.

**Comment by heltonteixeira** [comment]
+1 on this
Even when using @ to a file, sometimes it completely ignores it content, specially if it's on the first message
